release-notes-agent/src/main/java/com/epam/release/notes/agent/service/action/mail/TemplateNotificationServiceImpl.java

```diff
@@ -17,15 +17,64 @@
 import com.epam.release.notes.agent.entity.github.GitHubIssue;
 import com.epam.release.notes.agent.entity.jira.JiraIssue;
 import com.epam.release.notes.agent.entity.mail.EmailContent;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
+import org.thymeleaf.context.Context;
+import org.thymeleaf.spring5.SpringTemplateEngine;
 
 import java.util.List;
+import java.util.Locale;
 
 @Service
-public class TemplateNotificationServiceImpl implements TemplateNotificationService {
+public class TemplateNotificationServiceImpl implements TemplateNotificationService{
+
+    private final SpringTemplateEngine templateEngine;
+    private final String emailToAdminTemplateName;
+    private final String emailToSubscribersTemplateName;
+    private final String emailToAdminTitle;
+    private final String emailToSubscribersTitle;
+
+    public TemplateNotificationServiceImpl(
```

Comment:
since this implementation specifically about `email` let's name it `EmailTemplateNotificationService`